### ISSUE TITLE
Pylance error detection does not update in Jupyter Notebook

### ISSUE THREAD

**Issue by nothead31** [microsoft/pylance-release#1509]
<h3> Environment data </h3>

-   Language Server version: 2021.6.4-pre.1
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): Anaconda 3 (Jupyter notebook)
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: off
- VS-Code Insiders

<h3> Actual behaviour </h3>

Pylance does not update syntax error detection with open file. In the attached code snippet, Pylance showed 22 errors based on various typing mistakes when writing, and did not clear these errors until the Jupyter Notebook was closed and restarted. Errors included notifying of typing mistakes, missing semicolons, and undeclared variables.

<h3> Expected behavior </h3>

Pylance should be able to register when an error is corrected in a Jupyter notebook and update the Problems or Errors section of the interface as neeed.

<h3> Code Snippet / Additional information </h3>

```

with open('insurance.csv') as file:
    data = csv.DictReader(file);
    for i in data:
        dataset.append(i);
```

**Comment by jakebailey** [comment]
I don't think that this is a Pylance bug; the actual sending of changes is handled by the Jupyter extension and core extension. If this is done correctly, the diagnostics will get refreshed.

**Comment by jakebailey** [comment]
Can you switch your language server setting to JediLSP or Microsoft and see if things change?

**Comment by nothead31** [comment]
Doing so caused other issues with Jupyter, but did fix this issue, so the problem lies there. I'll reinstall jupyter and investigate from that angle.

**Comment by jakebailey** [comment]
Well, I'd think that if changing the language server meant this was no longer reproducible, then it would then be a bug in Pylance (which of course is possible), but maybe I'm misunderstanding you.

**Comment by nothead31** [comment]
Changing the language server crashed Jupyter. Once I went back in and messed with the settings, the default language server was working fine again, which indicates to me that the issue was something with Jupyter. I switched back to the default language server and tried  to reproduce, but the error could not be replicated.

In other words: restarting Jupyter fixed the issue, changing the language server had no impact on resolution.
